### PR TITLE
gh-93649: Split vectorcall testing from _testcapimodule.c

### PR DESCRIPTION
**Pull request by encukou**
The `_testcapimodule.c` file is getting too large to work with effectively.
This PR lays out a general structure of how tests can be split up, with more splitting to come later if the structure is OK.

Vectorcall tests aren't the biggest issue -- it's just an area I want to work on next, so I'm starting here.
An issue specific to vectorcall tests is that it wasn't clear that e.g. `MethodDescriptor2` is related to testing vectorcall: the `/* Test PEP 590 */` section had an ambiguous end. Separate file should make things like this much clearer.
OTOH, for some pieces it might not be clear where they should be -- I left `meth_fastcall` with tests of the other calling conventions. IMO, even with the ambiguity it's still worth it to split the huge file up.

I'm not sure about the buildsystem changes, hopefully CI will tell me what's wrong.

@vstinner, @markshannon: Do you think this is a good idea?

* Issue: gh-93649

Automerge-Triggered-By: GH:encukou

### CODE REVIEW

File: PCbuild/_testcapi.vcxproj [hunk 1]
```diff
@@ -107,4 +108,4 @@
   <Import Project="$(VCTargetsPath)\Microsoft.Cpp.targets" />
   <ImportGroup Label="ExtensionTargets">
   </ImportGroup>
-</Project>
```
Comment (erlend-aasland) [inline PCbuild/_testcapi.vcxproj]: I believe you'll have to revert the added newlines in the `PCbuild` directory.

**Review by erlend-aasland — COMMENTED** [review]
This is IMO a nice improvement. Thanks!

You might want to add `Modules/_testcapi/testdcapimodule_parts.h` to `MODULE__TESTCAPI_DEPS` in `Makefile.pre.in`.

**Comment by vstinner** [comment]
I suggest shorter file names:

* ``Modules/_testcapi/test_vectorcall.c`` -> ``Modules/_testcapi/vectorcall.c``
* ``Modules/_testcapi/testcapimodule_parts.h`` -> ``Modules/_testcapi/parts.h``

Would it be possible to move ``Modules/_testcapimodule.c`` to ``Modules/_testcapi/module.c``?

**Comment by vstinner** [comment]
> @vstinner, @markshannon: Do you think this is a good idea?

Yes, I like the idea of splitting the long _testcapimodule.c into multiple shorter C files.

**Comment by encukou** [comment]
> Would it be possible to move Modules/_testcapimodule.c to Modules/_testcapi/module.c?

In a different PR, maybe. There are a few more files to go on that ride, and I'd like to limit  this PR to one change.

File: Modules/_testcapimodule.c [hunk 2]
```diff
@@ -7973,6 +7728,11 @@ PyInit__testcapi(void)
                            (PyObject *) &ContainerNoGC_type) < 0)
         return NULL;
 
+    /* Include tests from the _testcapi/ directory */
+    if(_PyTestCapi_Init_Vectorcall(m) < 0) {
```
Comment (vstinner) [inline Modules/_testcapimodule.c]: nitpick:

```suggestion
    if (_PyTestCapi_Init_Vectorcall(m) < 0) {
```

**Review by vstinner — APPROVED** [review]
LGTM.

While _testcapimodule.c is built as a single C extension, I'm not convinced by the purpose of _testcapi/parts.h which only declares a single function. But it's more explicit for you, go for it.

**Comment by encukou** [comment]
Thanks!

**Comment by encukou** [comment]
_testcapi/parts.h should list the init functions from all the parts. There'll be more :)

**Comment by tiran** [comment]
The PR broke WASM build builts:

https://buildbot.python.org/all/#/builders/1050/builds/86

```
error: unable to open output file 'Modules/_testcapi/vectorcall.o': 'No such file or directory'
1 error generated.
emcc: error: 'ccache /opt/emsdk/upstream/bin/clang -target wasm32-unknown-emscripten -D__EMSCRIPTEN_SHARED_MEMORY__=1 -DEMSCRIPTEN -D__EMSCRIPTEN_major__=3 -D__EMSCRIPTEN_minor__=1 -D__EMSCRIPTEN_tiny__=15 -fignore-exceptions -mllvm -combiner-global-alias-analysis=false -mllvm -enable-emscripten-sjlj -mllvm -disable-lsr -Werror=implicit-function-declaration -Xclang -iwithsysroot/include/SDL --sysroot=/opt/buildbot/.emscripten_cache/sysroot -Xclang -iwithsysroot/include/compat -Wsign-compare -Wunreachable-code -DNDEBUG -g3 -fwrapv -O3 -Wall -pthread -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Werror=implicit-function-declaration -fvisibility=hidden -I../../Include/internal -IObjects -IInclude -IPython -I. -I../../Include -DPy_BUILD_CORE_BUILTIN -c -matomics -mbulk-memory ../../Modules/_testcapi/vectorcall.c -o Modules/_testcapi/vectorcall.o' failed (returned 1)
make: *** [Makefile:2800: Modules/_testcapi/vectorcall.o] Error 1
make: *** Waiting for unfinished jobs....
emmake: error: 'make -j2 all' failed (returned 2)
```

gh-94695 should fix them

**Pull request merged**